pandas/core/series.py

```diff
@@ -158,6 +158,8 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
 
             if data is None:
                 data = {}
+                if index is None:
```

Comment:
Should I add the test code in the following test? Or somewhere else?
Is setting `check_index_type=True` enough?
```python
def test_constructor_empty(self):
        empty = Series()
        empty2 = Series([])

        # the are Index() and RangeIndex() which don't compare type equal
        # but are just .equals
        assert_series_equal(empty, empty2, check_index_type=False)

        empty = Series(index=lrange(10))
        empty2 = Series(np.nan, index=lrange(10))
        assert_series_equal(empty, empty2)
```